File: Wheels-n-Deals/Wheels-n-Deals.API/DataLayer/Entities/Announcement.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Wheels_n_Deals.API.DataLayer.Entities;

public class Announcement : BaseEntity
{
    public User? User { get; set; }

    public Vehicle? Vehicle { get; set; }

    [MaxLength(150)]public string Title { get; set; } = string.Empty;

    [MaxLength(1000)] public string Description { get; set; } = string.Empty;

    [MaxLength(50)] public string County { get; set; } = string.Empty;

    [MaxLength(50)] public string City { get; set; } = string.Empty;

    public List<Image>? ImagesUrl { get; set; }
}

File: Wheels-n-Deals/src/Wheels-n-Deals.Frontend/src/app/Models/announcement.ts
import { User } from './user';
import { Vehicle } from './vehicle';
export interface Announcement {
  id: string,
  title: string,
  description: string,
  county: string,
  city: string,
  dateCreated: string,
  dateModified: string,
  images: string[],
  vehicle: Vehicle
  user: User
}
